tests/integration/ceph_csi_test.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/rook/rook/tests/framework/clients"
+	"github.com/rook/rook/tests/framework/installer"
+	"github.com/rook/rook/tests/framework/utils"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	csiSecretName        = "ceph-csi-secret"
+	csiSCRBD             = "ceph-csi-rbd"
+	csiSCCephFS          = "ceph-csi-cephfs"
+	csiPoolRBD           = "csi-rbd"
+	csiPoolCephFS        = "csi-cephfs"
+	csiTestRBDPodName    = "csi-test-rbd"
+	csiTestCephFSPodName = "csi-test-cephfs"
+)
+
+// a minimal mon status struct that contains monmap
+type cephMonStatus struct {
+	Name   string `json:"name"`
+	Monmap struct {
+		Mons []struct {
+			Rank       int    `json:"rank,omitempty"`
```

Comment:
@rootfs would be good if we can document about these fields in source .

Reply:
this is for the purpose of unmarshall mon status, the actual meaning of the field may be found in ceph doc, I'll add it.

http://docs.ceph.com/docs/mimic/rados/operations/monitoring/

Reply:
@rootfs would it work to use the definition rook has for [MonStatusResponse](https://github.com/rook/rook/blob/master/pkg/daemon/ceph/client/mon.go#L27) for the monmap, or is this different?